lib/puppet_x/relay/agent/backend/orchestrator.rb

```diff
@@ -45,7 +45,14 @@ def exec(run, _state_dir, schedule)
               message: e.message,
               body: e.response.body,
             })
-            raise
+            run_results = {
```

Comment:
The set of HTTP error codes you're capturing here are probably way larger than what we should realistically immediately return to the user. I think we should probably `rescue Net::HTTPClientError` (https://github.com/ruby/ruby/blob/master/lib/net/http/responses.rb#L20) in a separate block (above this one), which will let us explicitly handle the permanent rejections the orchestrator API sends back, instead of e.g. coincidental 5xx errors or proxy errors. Then just leave this block as-is so it can retry other HTTP errors.

Reply:
@impl By "leaving the block as-is" you mean as it was before the PR? So it raises? If I understand correctly, it should raise StandardError for the job to be retried?

Reply:
I think almost all exceptions (except maybe Python-style keyboard interrupt) inherit StandardError so these would all be caught there.